### PR TITLE
Improve Linux installation instructions: apt

### PR DESCRIPTION
**Pull request by mikkoi**
New APT security model allows public keys to be connected only to specific repositories. This eliminates the problem that any key could be used in conjunction with any repository, not just the one which distributed its key.

"[..] The key MUST NOT be placed in /etc/apt/trusted.gpg.d or loaded by apt-key add. If future updates to the key will be managed by an apt/dpkg package as recommended below, then it SHOULD be downloaded into /usr/share/keyrings using the same filename that will be provided by the package. If it will be managed locally , it SHOULD be downloaded into /etc/apt/keyrings instead."

[OpenPGP Key Distribution](https://wiki.debian.org/DebianRepository/UseThirdParty#OpenPGP_Key_distribution)

https://manpages.debian.org/bullseye/apt/sources.list.5.en.html https://wiki.debian.org/DebianRepository/UseThirdParty

### CODE REVIEW

**Review by ChaseCrum — APPROVED** [review]
looks good to me - verified in labs and documentation

**Comment by jiasli** [comment]
One problem here: What if a user has already added the signing key as `/etc/apt/trusted.gpg.d/microsoft.gpg`? Do we need to instruct the user to move `/etc/apt/trusted.gpg.d/microsoft.gpg` to `/etc/apt/keyrings/microsoft.gpg` instead, and add `signed-by=/etc/apt/keyrings/microsoft.gpg` to `/etc/apt/sources.list.d/azure-cli.list`?

**Comment by mikkoi** [comment]
> One problem here: What if a user has already added the signing key as `/etc/apt/trusted.gpg.d/microsoft.gpg`? Do we need to instruct the user to move `/etc/apt/trusted.gpg.d/microsoft.gpg` to `/etc/apt/keyrings/microsoft.gpg` and add `signed-by=/etc/apt/keyrings/microsoft.gpg` to `/etc/apt/sources.list.d/azure-cli.list`?

No need to instruct. If user has earlier created a key in `/etc/apt/trusted.gpg.d/microsoft.gpg`, this key will not be used by the new `/etc/apt/sources.list.d/azure-cli.list` because it now has the `signed-by=` pointing it to the right key. The old key will remain but it won't be used by `apt`.

File: docs-ref-conceptual/includes/cli-install-linux-apt.md [hunk 1]
```diff
@@ -52,16 +52,18 @@ If you prefer a step-by-step installation process, complete the following steps
 2. Download and install the Microsoft signing key:
 
     ```bash
+    sudo mkdir -p /etc/apt/keyrings
     curl -sL https://packages.microsoft.com/keys/microsoft.asc |
         gpg --dearmor |
-        sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
+        sudo tee /etc/apt/keyrings/microsoft.gpg > /dev/null
+    sudo chmod go+r /etc/apt/keyrings/microsoft.gpg
```
Comment (bebound) [inline docs-ref-conceptual/includes/cli-install-linux-apt.md:59]: `sudo chmod go+r /etc/apt/keyrings/microsoft.gpg` seems to be a no-op.

``` bash
curl -sL https://packages.microsoft.com/keys/microsoft.asc | sudo gpg --dearmor -o ./kk.gpg
ll kk.gpg
-rw-r--r-- 1 root root 641 Apr 11 10:20 kk.gpg
```

Comment (Jonher937) [reply docs-ref-conceptual/includes/cli-install-linux-apt.md:59]: Wouldn't be with a modified `umask`. Explicitly setting it to `go+r` will however guarantee it's readable.

**Pull request merged**